pkg/resource/v1/certconfig/resource.go

```diff
@@ -3,6 +3,7 @@ package certconfig
 import (
 	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
 	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	certconfigkey "github.com/giantswarm/cert-operator/service/certconfig/v2/key"
```

Comment:
I find this still super weird. IMO there should be a key package for the versioned resource because I still think the versioning is important here. Using a resource version X with another versioned Y functionality from other packages is wrong. That makes the whole versioning pointless. I also wonder how the maintenance will work out when this here stays as v1 forever and other develop in different cycles. We already mix v1 with v2 which indicates for me we are doing something wrong here. That validates my opinion that we should just keep everything versioned in `github.com/giantswarm/cert-operator/service/certconfig/vX` because this common resource thing smells because of the mentioned reasons.

Reply:
Did you accidentally mix `cluster-operator` and `cert-operator` here? I think we have a new aspect of versioning in terms of how it's "used" in `cluster-operator`. In this case, I'm extracting information from `CertConfig` which is "managed" by `cert-operator` and therefore it's an interface of `cert-operator`. My sense made me to use `key` package from `cert-operator` because I thought that is the ultimate location of knowledge for migrating its custom object. Initially I was about to implement this functionality in `key` package in `cluster-operator`, but due to that reasoning, I choose to import it.

Reply:
I did indeed not realise that the import is `github.com/giantswarm/cert-operator`. We should definitely not import operators in operators. That makes no sense to me at all. And my argument towards the version mixing and weird imports still holds true in this situation.

Reply:
Agreed we shouldn't import operators anywhere. They aren't designed for that. Implementing `VersionBundleVersion` inside this codebase shouldn't be a big deal, especially when apiextensions repo is imported.